packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -2467,7 +2476,48 @@ export class CacheDataClient implements IDataClient {
         },
         (err, resp) => {
           if (resp?.found) {
-            resolve(new CacheDictionaryFetch.Hit(resp.found.items));
+            if (!options?.decompress) {
+              resolve(new CacheDictionaryFetch.Hit(resp.found.items));
+            } else {
+              if (this.valueCompressor === undefined) {
+                resolve(
+                  new CacheDictionaryFetch.Error(
+                    new InvalidArgumentError(
+                      'Compressor is not set, but `CacheClient.dictionaryFetch` was called with the `decompress` option; please install @gomomento/sdk-nodejs-compression and call `Configuration.withCompressionStrategy` to enable compression.'
+                    )
+                  )
+                );
+              } else {
+                const decompressedItemPromises = resp.found.items.map(item => {
+                  // This check shouldn't be necessary given the one in the outer scope,
+                  // but the TypeScript compiler doesn't seem to understand that.
+                  if (this.valueCompressor === undefined) {
+                    throw new InvalidArgumentError(
+                      'Compressor is not set, but `CacheClient.dictionaryFetch` was called with the `decompress` option; please install @gomomento/sdk-nodejs-compression and call `Configuration.withCompressionStrategy` to enable compression.'
+                    );
+                  }
+
+                  return this.valueCompressor
+                    .decompressIfCompressed(item.value)
+                    .then(v => {
+                      item.value = v;
+                      return item;
+                    })
+                    .catch(e => {
+                      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                      throw new InvalidArgumentError(`${e}`);
```

Comment:
Cant we just wrap caught error? Seems weird to me to cast as string here.

Reply:
I had to do this because `e` is `any` in this context. Can revisit later!